middleware/src/services/peers.ts

```diff
@@ -0,0 +1,57 @@
+import axios from "axios";
+
+import { Peer, StatusResponse } from "../types";
+import { getPeers, setPeers } from "./cache";
+
+export const checkPeerIsAlive = async (ip: string) => {
+  // Check if given ip is v4 ip
+  if (
+    /^(?:25[0-5]|2[0-4]\\d|1\\d\\d|[1-9]\\d|\\d)(?:\\.(?:25[0-5]|2[0-4]\\d|1\\d\\d|[1-9]\\d|\\d)){3}$/.test(
+      ip
+    )
+  ) {
+    throw Error(`Invalid IP address: ${ip}`);
+  }
+  try {
+    await axios.post(`http://${ip}:7777/rpc`);
+
+    return true;
+  } catch (error) {
+    console.error(error);
+    return false;
+  }
+};
+
+export const fetchPeers = async (update = false): Promise<StatusResponse> => {
+  // TODO: Replace hardcoded
+  const node = "http://65.21.229.213:8888/status";
```

Comment:
why there is a hardcoded node URL? 
for now it should be one taken from `nodeManager.getActiveNode();`, right?

Reply:
and in future, indeed we can replace it with loadbalancer as @rob-casper suggested.

Reply:
oh, I see you're using the REST API, why not the rpc ? you can use `CasperServiceByJsonRPC` and `cc.getPeers()`